sdk/core/azure-core/src/main/java/com/azure/core/util/polling/SyncPoller.java

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.polling;
+import java.time.Duration;
+
+/**
+ * Type for doing synchronous polling of a long-running operation.
+ *
+ * @param <T> The type of poll response value.
+ * @param <U> The type of the final result of long-running operation.
+ */
+public interface SyncPoller<T, U> {
+    /**
+     * Poll once and return the poll response received.
+     *
+     * @return the poll response
+     */
+    PollResponse<T> poll();
+
+    /**
+     * Wait for polling to complete. The polling is considered complete based on status defined in
+     * {@link LongRunningOperationStatus}.
+     *
+     * @return the final poll response
+     */
+    PollResponse<T> waitForCompletion();
+
+    /**
+     * Wait for polling to complete with a timeout. The polling is considered complete based on
+     * status defined in {@link LongRunningOperationStatus}.
+     *
+     * @param timeout the duration to waits for polling completion.
+     * @return the final poll response
+     * @throws java.util.concurrent.TimeoutException TimeoutException will be thrown if polling
+     *         does not complete within the given {@link Duration}.
+     */
+    PollResponse<T> waitForCompletion(Duration timeout);
+
+    /**
+     * Wait until the given {@link LongRunningOperationStatus} is received.
+     *
+     * @param statusToWaitFor the desired {@link LongRunningOperationStatus} to block for.
+     * @return {@link PollResponse} whose {@link PollResponse#getStatus()} matches {@code statusToWaitFor}.
+     * @throws IllegalArgumentException if {@code timeout} is zero or negative and if {@code statusToWaitFor} is
+     * {@code null}.
+     */
+    PollResponse<T> waitUntil(LongRunningOperationStatus statusToWaitFor);
+
+    /**
+     * Wait until the given {@link LongRunningOperationStatus} is received.
+     *
+     * @param statusToWaitFor the desired {@link LongRunningOperationStatus} to block for.
+     * @param timeout the duration to waits for the polling.
+     * @return {@link PollResponse} whose {@link PollResponse#getStatus()} matches {@code statusToWaitFor}.
+     * @throws java.util.concurrent.TimeoutException TimeoutException will be thrown if polling
+     *         does not find the matching status within the given {@link Duration}.
+     */
+    PollResponse<T> waitUntil(LongRunningOperationStatus statusToWaitFor, Duration timeout);
```

Comment:
Worth considering `waitUntilStatusOrCompletion(LongRunningOperationStatus statusToWaitFor)` overload which waits until the requested status is reached or the operation is completed, whichever is earlier.

Reply:
nice, let's add it.

Reply:
Or `waitUtilStatuses(longRunningOperationStatus... statustoWaitFor)`?

Anyways, I'm joking 😆 But I think there's a limit to how many of these convenience methods to add. Users can always turn to the async API 😄

Reply:
park for discussion tomorrow

Reply:
We could just keep `waitUtil(longRunningOperationStatus)` and when needed we can make it var-args whiteout breaking any breaking change. This also means, the overload for timeout should take timeout as first arg so that we can turn second param var-args without breaking.

yes, let's discuss this.

Reply:
Ok, I made Timeout first parameter so when needed we can support var-args without breaking.